creds/ssm/ssm.go

```diff
@@ -0,0 +1,113 @@
+package ssm
+
+import (
+	"bytes"
+	"strings"
+	"text/template"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/aws/aws-sdk-go/service/ssm/ssmiface"
+	varTemplate "github.com/cloudfoundry/bosh-cli/director/template"
+)
+
+type Ssm struct {
+	log             lager.Logger
+	api             ssmiface.SSMAPI
+	TeamName        string
+	PipelineName    string
+	SecretTemplates []*template.Template
+}
+
+func NewSsm(log lager.Logger, api ssmiface.SSMAPI, teamName string, pipelineName string, secretTemplates []*template.Template) *Ssm {
+	return &Ssm{
+		log:             log,
+		api:             api,
+		TeamName:        teamName,
+		PipelineName:    pipelineName,
+		SecretTemplates: secretTemplates,
+	}
+}
+
+func (s *Ssm) buildSecretName(nameTemplate *template.Template, varName string) (string, error) {
+	var buf bytes.Buffer
+	err := nameTemplate.Execute(&buf, &SsmSecret{
+		Team:     s.TeamName,
+		Pipeline: s.PipelineName,
+		Secret:   varName,
+	})
+	return buf.String(), err
+}
+
+func (s *Ssm) Get(varDef varTemplate.VariableDefinition) (interface{}, bool, error) {
+	for _, st := range s.SecretTemplates {
+		if secret, err := s.buildSecretName(st, varDef.Name); err != nil {
+			s.log.Error("Failed to build SSM parameter path from secret", err, lager.Data{
+				"template": st.Name(),
+				"secret":   varDef.Name,
+			})
+			return nil, false, err
+		} else if value, found, err := s.getParameterByName(secret); err != nil {
+			s.log.Error("Failed to get SSM paramter by name", err, lager.Data{
+				"template": st.Name(),
+				"secret":   secret,
+			})
+			return nil, false, err
+		} else if found {
```

Comment:
This chaining of `else-if`s feels a bit unconventional. I would just move the `.getXXX` calls out of the `if`-block, since it returns other important values aside from the `err`:

```go
value, found, err = s.getXXX();
if err != nil {
  // ...
}

if found {
  return value, true, nil
}

value, found, err = s.getXXX()
// ...
```